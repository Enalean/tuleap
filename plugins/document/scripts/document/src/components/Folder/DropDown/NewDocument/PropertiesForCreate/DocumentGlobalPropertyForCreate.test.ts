/*
 * Copyright (c) Enalean 2022 -  Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import type { VueWrapper } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import DocumentGlobalPropertyForCreate from "./DocumentGlobalPropertyForCreate.vue";
import type { Folder, ItemFile } from "../../../../../type";

describe("DocumentGlobalPropertyForCreate", () => {
    function createWrapper(
        item: ItemFile,
        parent: Folder,
    ): VueWrapper<InstanceType<typeof DocumentGlobalPropertyForCreate>> {
        return shallowMount(DocumentGlobalPropertyForCreate, {
            props: { currentlyUpdatedItem: item, parent },
        });
    }

    it(`renders component`, () => {
        const item = {
            id: 1,
            title: "My document",
            status: "Draft",
            description: "A custom description",
        } as ItemFile;
        const parent = {
            id: 2,
        } as Folder;
        const wrapper = createWrapper(item, parent);

        expect(wrapper.element).toMatchSnapshot();
    });
});
