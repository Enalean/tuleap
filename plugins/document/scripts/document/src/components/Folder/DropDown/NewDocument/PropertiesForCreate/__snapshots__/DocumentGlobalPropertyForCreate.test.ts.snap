// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentGlobalPropertyForCreate renders component 1`] = `
<div
  class="document-properties"
>
  <div
    class="document-properties-title-and-status-container"
  >
    <title-property-stub
      currentlyupdateditem="[object Object]"
      isinupdatecontext="false"
      parent="[object Object]"
      value="My document"
    />
    <status-property-with-custom-binding-for-document-create-stub
      parent="[object Object]"
      status_value="Draft"
    />
  </div>
  <preview-filename-stub
    item="[object Object]"
  />
  <description-property-stub
    value="A custom description"
  />
  
  
</div>
`;
